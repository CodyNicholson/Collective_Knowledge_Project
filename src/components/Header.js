import React from 'react';
import { Link } from 'react-router-dom';

const LoggedOutView = props => {
  if (!props.currentUser) {
    return (
      <ul className="nav navbar-nav pull-xs-right">
      

        <li className="nav-item">
          <Link to="/" className="nav-link">
          <i className="ion-home"></i>&nbsp;Home
          </Link>
        </li>

        <li className="nav-item">
          <Link to="/DevNotes" className="nav-link">
          <i className="ion-home"></i>&nbsp;DevNotes
          </Link>
        </li>

        <li className="nav-item">
          <Link to="/about" className="nav-link">
            <i className="ion-information-circled"></i>&nbsp;About
          </Link>
        </li>

        <li className="nav-item">
          <Link to="/login" className="nav-link">
            Sign in
          </Link>
        </li>

        <li className="nav-item">
          <Link to="/register" className="nav-link">
            Sign up
          </Link>
        </li>

      </ul>
    );
  }
  return null;
};

const LoggedInView = props => {
  if (props.currentUser) {
    return (
      <ul className="nav navbar-nav pull-xs-right">


        <li className="nav-item">
          <Link to="/" className="nav-link">
          <i className="ion-home"></i>&nbsp;Home
          </Link>
        </li>

        <li className="nav-item">
          <Link to="/about" className="nav-link">
            <i className="ion-information-circled"></i>&nbsp;About
          </Link>
        </li>

        <li className="nav-item">
          <Link to="/editor" className="nav-link">
            <i className="ion-compose"></i>&nbsp;New Post
          </Link>
        </li>

        <div className="w3-dropdown-hover">
          <button className="nav-link">{props.currentUser.username}</button>
          <div className="w3-dropdown-content w3-bar-block w3-border">
            <a className="nav-item profile">
              <Link to={`/@${props.currentUser.username}`} className="nav-link">
                {/* <img src={props.currentUser.image} className="user-pic" /> */}
                <i className="ion-ios-contact-outline"></i>&nbsp;
                Profile
              </Link>
            </a>
            <a className="nav-item">
              <Link to="/settings" className="nav-link">
                <i className="ion-gear-a"></i>&nbsp;Settings
              </Link>
            </a>
            <a className="nav-item">
              <Link to="/settings" className="nav-link">
                <i className="ion-ios-undo"></i>&nbsp;Logout
              </Link>
            </a>
          </div>
        </div>
      </ul>
    );
  }

  return null;
};

class Header extends React.Component {
  render() {
    return (
      <nav className="navbar navbar-light">
        <div className="container">

          <Link to="/" className="navbar-brand">
            {this.props.appName}
          </Link>

          <LoggedOutView currentUser={this.props.currentUser} />

          <LoggedInView currentUser={this.props.currentUser} />

        </div>
      </nav>
    );
  }
}

export default Header;
