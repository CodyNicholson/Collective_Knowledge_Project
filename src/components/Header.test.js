//This file was begun using code from Link.react.test.js, from the Jest Testing React Apps Guide
//https://facebook.github.io/jest/docs/en/tutorial-react.html
import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

test('Tags module test', () => {
  const component = renderer.create(
    <MemoryRouter>
    	<Header>
    	</Header>
    </MemoryRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});