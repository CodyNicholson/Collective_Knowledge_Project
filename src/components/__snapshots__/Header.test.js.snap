// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags module test 1`] = `
<nav
  className="navbar navbar-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    />
    <ul
      className="nav navbar-nav pull-xs-right"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          <i
            className="ion-home"
          />
           Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          <i
            className="ion-information-circled"
          />
           About
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/DevNotes"
          onClick={[Function]}
        >
          <i
            className="ion-ios-paper"
          />
           DevNotes
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/sponsors"
          onClick={[Function]}
        >
          <i
            className="ion-ios-people"
          />
           Sponsors
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/register"
          onClick={[Function]}
        >
          Sign up
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
